### ISSUE TITLE
Unable to build with CC="clang -std=gnu99"

### ISSUE THREAD

**Issue by dholbert** [alexcrichton/cc-rs#300]
STR:
Build some crate using cc-rs, using environmental variable CC="clang -std=gnu99". E.g. run the following commands:
````
mkdir /tmp/my-project
cd /tmp/my-project
cargo init
echo libloading = \"0.5\" | tee -a Cargo.toml
CC="clang -std=gnu99" cargo build
````

EXPECTED RESULTS: Build success.
ACTUAL RESULTS: Build error with the following output:
````
error: failed to run custom build command for `libloading v0.5.0`
process didn't exit successfully: `/tmp/my-project/target/debug/build/libloading-58cd8ebbf12e35d8/build-script-build` (exit code: 101)
--- stdout
cargo:rustc-link-lib=dl
TARGET = Some("x86_64-unknown-linux-gnu")
OPT_LEVEL = Some("0")
TARGET = Some("x86_64-unknown-linux-gnu")
HOST = Some("x86_64-unknown-linux-gnu")
TARGET = Some("x86_64-unknown-linux-gnu")
TARGET = Some("x86_64-unknown-linux-gnu")
HOST = Some("x86_64-unknown-linux-gnu")
CC_x86_64-unknown-linux-gnu = None
CC_x86_64_unknown_linux_gnu = None
HOST_CC = None
CC = Some("clang -std=gnu99")
TARGET = Some("x86_64-unknown-linux-gnu")
HOST = Some("x86_64-unknown-linux-gnu")
CFLAGS_x86_64-unknown-linux-gnu = None
CFLAGS_x86_64_unknown_linux_gnu = None
HOST_CFLAGS = None
CFLAGS = None
DEBUG = Some("true")
running: "clang -std=gnu99" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-g" "--target=x86_64-unknown-linux-gnu" "-Wall" "-Wextra" "-o" "/tmp/my-project/target/debug/build/libloading-823a806b14f98686/out/src/os/unix/global_static.o" "-c" "src/os/unix/global_static.c"

--- stderr
thread 'main' panicked at '

Internal error occurred: Failed to find tool. Is `clang -std=gnu99` installed?
````

**Comment by dholbert** [comment]
I run into this when building mozilla-central, if I have "export CC=clang" in my environment (e.g. set via .bash_aliases), as discussed in https://bugzilla.mozilla.org/show_bug.cgi?id=1445528 .  Some internal mozilla build tool appends "-std=gnu99" to $CC before we compile libloading and so `cc-rs` ends up seeing CC=" clang -std=gnu99"

**Comment by ishitatsuyuki** [comment]
Had this improved as of 1.0.8?

**Comment by dholbert** [comment]
Nope, I still get same results from my STR in the initial comment here (after "Downloading cc v1.0.8" in the terminal output).

**Comment by sylvestre** [comment]
Alex told me that I reported the same bug here: https://bugzilla.mozilla.org/show_bug.cgi?id=1447660

**Comment by alexcrichton** [comment]
@dholbert or @sylvestre can y'all try out master and see if that works?

**Comment by dholbert** [comment]
Seems to work! My STR to confirm that it was working:
````
mkdir /tmp/my-project
cd /tmp/my-project
cargo init

echo "libloading = \"0.5\"
[patch.crates-io]
cc = { git = 'https://github.com/alexcrichton/cc-rs' }
" | tee -a Cargo.toml

CC="clang -std=gnu99" cargo build
````

I tried a few trickier CC values too, to combine this with #297, #298, and #299:
`CC="ccache clang -std=gnu99"` and `CC=" icecc clang -std=gnu99"`, and I didn't hit any errors.

**Comment by dholbert** [comment]
Hopefully this will fix https://bugzilla.mozilla.org/show_bug.cgi?id=1445528 , once we pull in the updated version of this library!

(I'm not actually sure how to test this in the context of mozilla-central.  But I'm happy to give that a shot if it's easy. It looks like our copy of this library lives at https://hg.mozilla.org/mozilla-central/annotate/tip/third_party/rust/cc/Cargo.toml#l15 and is at version 1.0.4 right now, so it's a bit out of date.)

**Comment by alexcrichton** [comment]
Ok thanks for checking! I've published 1.0.9
